lxc/list.go

```diff
@@ -781,3 +819,112 @@ func (c *cmdList) NumberOfProcessesColumnData(cInfo api.InstanceFull) string {
 func (c *cmdList) locationColumnData(cInfo api.InstanceFull) string {
 	return cInfo.Location
 }
+
+func matchByType(cInfo *api.Instance, query string) bool {
```

Comment:
For this function and all the ones below that you're introducing, can you tie them to cmdList (`(c *cmdList`)?

I know they don't need the struct's state so can be standalone but we try to avoid that as a pattern in the `lxc` package as it can quickly start conflicting with other sub-commands.